### PR TITLE
Update recipe to use the latest version.

### PR DESCRIPTION
**Pull request by matham**
I'm copying the relevant sections from the anaconda build source, which already builds the latest version. Hopefully the CI will all work.

### CODE REVIEW

**Comment by ccordoba12** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,29 +1,42 @@
...
 
 source:
-  fn: gstreamer-{{ version }}.tar.xz
   url: https://gstreamer.freedesktop.org/src/gstreamer/gstreamer-{{ version }}.tar.xz
   sha256: {{ sha256 }}
 
 build:
-  skip: true  # [win or osx]
-  number: 2
+  skip: true  # [win]
```
Comment (ccordoba12) [inline recipe/meta.yaml]: Why this shouldn't be skipped for macOS?

**Comment by matham** [comment]
I copied it over from the anaconda recipe which builds it for osx and
linux. So its supposed to work on osx as well. I guess something is wrong
with how I transcribed it.

File: recipe/install_gstreamer.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
```
Comment (ccordoba12) [inline recipe/install_gstreamer.sh]: Please rename this to `build.sh`.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,29 +1,42 @@
...
+  number: 0
+  script: install_gstreamer.sh
```
Comment (ccordoba12) [inline recipe/meta.yaml]: Please remove this line. It seems the recipe in Anaconda is using a meta-recipe to build Gstreamer and its plugins, but that's not something we want here.

File: recipe/meta.yaml (hunk 1 above)
Comment (matham) [reply recipe/meta.yaml]: I copied this over from the anaconda recipes, which does support osx so I would like to leave it. Also, the main reason for the PR, is to add osx support (although it turns out I could pull it from the main anaconda repo, see https://github.com/conda-forge/staged-recipes/pull/5966). But I do admit that I don't have osx. Still, given that it worked in anaconda, it shouldn't be too hard to make it work here, I'm guessing.

**Comment by matham** [comment]
It managed to build on linux, but errored out on the test with 

```
gst-inspect-1.0: symbol lookup error: /home/conda/feedstock_root/build_artifacts/gstreamer_1527668447365/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/bin/../lib/libgstreamer-1.0.so.0: undefined symbol: g_cclosure_marshal_generic
```
On osx the tests failed with 
```
gst-inspect-1.0 --version
dyld: Library not loaded: @rpath/libgobject-2.0.0.dylib
  Referenced from: /Users/travis/miniconda3/conda-bld/gstreamer_1527668937920/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/bin/gst-inspect-1.0
  Reason: image not found
/Users/travis/miniconda3/conda-bld/gstreamer_1527668937920/test_tmp/run_test.sh: line 7: 20685 Trace/BPT trap: 5       gst-inspect-1.0 --version
```

I'm not really sure how to fix it.

**Comment by ccordoba12** [comment]
@matham, please restore the linux and osx selectors since Ray is going to take care of that.

**Comment by mariusvniekerk** [comment]
For this to work you need to add a few more things from `host` to `run` as well.  Conda-forge does not have run_exports for all upstream things yet.

**Comment by ocefpaf** [comment]
Thanks @matham. All the changes here should be covered in #17. Let me know if you still need anything from this PR though.

**Pull request closed without merging**